grails-app/controllers/org/pih/warehouse/api/ProductSupplierApiController.groovy

```diff
@@ -2,16 +2,16 @@ package org.pih.warehouse.api
 
 import grails.converters.JSON
 import org.pih.warehouse.data.ProductSupplierService
-import org.pih.warehouse.product.ProductSupplierListDto
+import org.pih.warehouse.product.ProductSupplier
 import org.pih.warehouse.product.ProductSupplierListParams
 
 class ProductSupplierApiController {
 
     ProductSupplierService productSupplierService
 
     def list(ProductSupplierListParams filterParams) {
-        List<ProductSupplierListDto> productSuppliers = productSupplierService.getProductSuppliers(filterParams)
-        render([data: productSuppliers, totalCount: productSuppliers.totalCount] as JSON)
+        List<ProductSupplier> productSuppliers = productSupplierService.getProductSuppliers(filterParams)
+        render([data: productSuppliers.collect { it.toJson() }, totalCount: productSuppliers.totalCount] as JSON)
```

Comment:
@jmiranda wanted me to revert my convention with the dto pattern and do it `.toJson()` way.

> I remember we were talking about it not too long ago but I am not sure if we want to move away from Marshalers and toJson() or are we still doing it?
@jmiranda said he would eventually like to get rid of the marshallers. 

I'm also against them, because by doing it `.toJson()` way, you immediately can see what will be returned, while with marshallers you can see in the method something like `return productSuppliers` which is `List<ProductSupplier>` and in the response you suddenly see some other dto.

Reply:
ok, got it